addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -9273,67 +9273,69 @@ QUnit.module("Views", (hooks) => {
             },
         };
 
-        var onchangeDef;
-        var createDef = makeDeferred();
-        await makeView({
+        let onchangeDef;
+        const createDef = makeDeferred();
+        const form = await makeView({
             type: "form",
             resModel: "partner",
             serverData,
-            arch: '<form><field name="display_name"/><field name="foo"/></form>',
-            mockRPC(route, args) {
-                var result = this._super.apply(this, arguments);
+            arch: `
+                <form>
+                    <field name="display_name"/>
+                    <field name="foo"/>
+                </form>`,
+            async mockRPC(route, args) {
                 if (args.method === "onchange") {
-                    return Promise.resolve(onchangeDef).then(_.constant(result));
+                    await onchangeDef;
                 }
                 if (args.method === "create") {
-                    return Promise.resolve(createDef).then(_.constant(result));
+                    await createDef;
                 }
-                return result;
             },
         });
 
         // edit foo to trigger a delayed onchange
         onchangeDef = makeDeferred();
-        await editInput(target, ".o_field_widget[name=foo]", "trigger onchange");
+        await editInput(target, ".o_field_widget[name=foo] input", "trigger onchange");
 
         assert.strictEqual(
-            target.querySelector(".o_field_widget[name=display_name]").value,
+            target.querySelector(".o_field_widget[name=display_name] input").value,
             "default"
         );
 
         // save (will wait for the onchange to return), and will be delayed as well
         await click(target.querySelector(".o_form_button_save"));
 
-        assert.hasClass(target.querySelector(".o_form_view"), "o_form_editable");
+        assert.containsOnce(target, ".o_form_editable");
         assert.strictEqual(
-            target.querySelector(".o_field_widget[name=display_name]").value,
+            target.querySelector(".o_field_widget[name=display_name] input").value,
             "default"
         );
 
         // simulate a click on the breadcrumbs to leave the form view
-        // form.canBeRemoved();
-        await testUtils.nextTick();
+        form.model.root.save({ noReload: true, stayInEdition: true });
```

Comment:
I would either do exactly what the comment says, so spawn a webclient (real integration test), or simulate what it does in our new implementation, that is triggering an event on the bus ("CLEAR-UNCOMMITTED-CHANGES")